src/org/traccar/protocol/NavisProtocolDecoder.java

```diff
@@ -272,41 +317,343 @@ private Object processArray(DeviceSession deviceSession, Channel channel, ByteBu
         return positions;
     }
 
-    private Object processHandshake(Channel channel, SocketAddress remoteAddress, ByteBuf buf) {
-        buf.readByte(); // colon symbol
-        if (getDeviceSession(channel, remoteAddress, buf.toString(StandardCharsets.US_ASCII)) != null) {
-            sendReply(channel, Unpooled.copiedBuffer("*<S", StandardCharsets.US_ASCII));
+    private boolean checkFlexBitfield(int index) {
+        int byteIndex = Math.floorDiv(index, 8);
+        int bitIndex = Math.floorMod(index, 8);
+        return BitUtil.check(flexBitfield[byteIndex], 7 - bitIndex);
+    }
+
+    private void skipFlexField(int index, ByteBuf buf) {
```

Comment:
This method seems unnecessary. It doesn't do much and used in a single place.

Reply:
Fixed